shared/array.c

```diff
@@ -37,7 +42,11 @@ int array_append(struct array *array, const void *element)
 	size_t idx;
 
 	if (array->count + 1 >= array->total) {
-		int r = array_realloc(array, array->total + array->step);
+		int r;
+
+		if (SIZE_MAX - array->total < array->step)
+			return -ENOMEM;
+		r = array_realloc(array, array->total + array->step);
```

Comment:
It's theoretical.

At one point I thought about turning `array->step` into an `unsigned char`, because callers of `array_init` can supply a custom step. If it's too large, it could overflow here. I eventually decided to stick with `size_t` for consistency.

No current caller of `array_init` could trigger this issue.